spec/models/post_spec.rb

```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+RSpec.describe Post do
+  before :all do
+    @author = User.create(name: 'Okari', photo: 'https://unsplash.com/photos/F_-0BxGuVvo',
+                          bio: 'Student from Microverse')
+  end
+
+  it 'is invalid without title' do
+    expect(Post.new(author: @author, text: 'Hello World!', comments_counter: 0, likes_counter: 0)).to_not be_valid
+  end
+
+  it 'is valid with existing title' do
+    expect(Post.new(author: @author, title: 'Okari Nyandika', text: 'Hello World!', comments_counter: 0,
+                    likes_counter: 0)).to be_valid
+  end
+
+  it 'is invalid with title exceeds 250' do
+    expect(Post.new(author: @author, title: 'Okari Nyandika' * 250, text: 'Hello World!', comments_counter: 0,
+                    likes_counter: 0)).to_not be_valid
+  end
+
+  it 'is valid if comments_counter is integer' do
+    expect(Post.new(author: @author, title: 'Okari Nyandika' * 250, text: 'Hello World!', comments_counter: 'one',
+                    likes_counter: 0)).to_not be_valid
+  end
+
+  it 'is invalid if likes_counter is less than zero' do
+    expect(Post.new(author: @author, title: 'Okari Nyandika' * 250, text: 'Hello World!', comments_counter: 1,
+                    likes_counter: -5)).to_not be_valid
+  end
+
+  context '#recent_comments' do
```

Comment:
- [ ] Please make sure to test all your methods in your User model for example the `recent_comments`, `update_posts_counter` methods. They should be called as you write their tests

Reply:
I have only one method in my User model so I really don’t understand which methods you are referring to, and the post model is full tested

Reply:
Hello @2004-okari , I meant to say the Post model. That was a typo on my part. Sorry about that. These are the methods seen in your post model:
![image](https://github.com/2004-okari/BLOG_APP/assets/45460192/26e8ea79-ca32-4c41-9328-14177336f35f)

Reply:
Hello @Lucas-Masaba ,
the 
```
 context '#update_posts_counter' do
    before :all do
      10.times do
        Post.create(author: @author, title: 'Kathy', text: 'Text here....', comments_counter: 0,
                    likes_counter: 0)
      end
    end

    it 'keeps track of posts and equals 10' do
      expect(@author.posts_counter).to eq 21
    end
  end```.

The #update_post_counter is a syntax for calling a method in the describe or context command. The recent post is a private method hence it cannot be tested hence I added a count_recent_post method that calls the private method hence it is tested indirectly